tests/integration/bond_test.py

```diff
@@ -265,3 +265,40 @@ def test_set_bond_mac_address(eth1_up):
         eth1_cur_state = current_state[INTERFACES][1]
         assert bond99_cur_state[Interface.MAC] == MAC1.upper()
         assert bond99_cur_state[Interface.MAC] == eth1_cur_state[Interface.MAC]
+
+
+@pytest.fixture
+def bond99():
+    with bond_interface(name='bond99', slaves=['eth1', 'eth2']) as bond_state:
+        yield bond_state
+
+
+def test_reordering_the_slaves_does_not_change_the_mac(bond99):
+    # Adding slaves to the bond
+    bond99[INTERFACES][0]['link-aggregation']['slaves'] = ['eth1', 'eth2']
```

Comment:
Bit the fixture has already done that for you, why are you doing it again? (including the apply part)